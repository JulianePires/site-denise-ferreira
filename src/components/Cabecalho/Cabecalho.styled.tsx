import cores from '@resources/cores'
import dispositivos from '@resources/dispositivos'
import margens from '@resources/margens'
import { detalhe, paragrafo, subtitulo } from '@resources/textos'
import styled from 'styled-components'

export const ContainerCabecalho = styled.header`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  @media ${dispositivos.tablet} {
    flex-direction: row;
  }

  width: 100%;
  height: 140px;
  padding: ${margens.xsmall}px ${margens.xxxlarge}px;

  background: ${cores.vinho};

  transition: 0.3s ease-in-out all;
`

export const ContainerOpcoesMenu = styled.ul`
  display: flex;
  align-items: center;
  gap: 20px;

  @media ${dispositivos.tablet} {
    gap: 40px;
  }

  padding: 0;

  list-style: none;
`

interface OpcaoMenuProps {
  atual: string
}

export const OpcaoMenu = styled.li<OpcaoMenuProps>`
  ${paragrafo}

  @media ${dispositivos.tablet} {
    font-family: ${detalhe.fontFamily};
    font-size: ${subtitulo.fontSize};
  }

  color: ${(props) => (props.atual === 'true' ? cores.laranja : cores.branco)};

  cursor: pointer;
  transition: 0.3s ease-in-out all;

  &:hover {
    color: ${cores.amarelo};
  }
`
