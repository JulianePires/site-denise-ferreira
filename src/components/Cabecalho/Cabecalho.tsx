import { Logo } from '@components/Logo';
import { ContainerCabecalho } from './Cabecalho.styled';

export function Cabecalho() {
  return (
    <ContainerCabecalho>
      <Logo />
    </ContainerCabecalho>
  );
}
