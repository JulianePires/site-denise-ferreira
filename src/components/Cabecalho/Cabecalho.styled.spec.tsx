import { OpcaoMenu } from './Cabecalho.styled'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('Estilo do CabeÃ§alho', () => {
  describe('OpcaoMenu', () => {
    it('DEVE sublinha quando a propriedade atual for "true"', () => {
      const tree = renderer.create(<OpcaoMenu atual="true" />).toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('text-decoration', 'underline')
    })
  })
})
