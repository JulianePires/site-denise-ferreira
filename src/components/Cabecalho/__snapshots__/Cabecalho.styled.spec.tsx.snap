// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Estilo do Cabe√ßalho OpcaoMenu DEVE sublinha quando a propriedade atual for "true" 1`] = `
.c0 {
  font-family: fontFamily;
  font-size: 18px;
  line-height: 25px;
  color: #ffffff;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  -webkit-transition: 0.3s ease-in-out all;
  transition: 0.3s ease-in-out all;
}

.c0:hover {
  color: #FFD65C;
}

@media (min-width:768px) {
  .c0 {
    font-family: fontFamily;
    font-size: 24px;
  }
}

<li
  className="c0"
/>
`;
